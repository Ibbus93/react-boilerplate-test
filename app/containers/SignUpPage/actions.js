/*
 *
 * SignUp actions
 *
 */

import { SIGN_UP_REQUEST, SIGN_UP_SUCCESS, SIGN_UP_ERROR } from './constants';

export function signUpRequest(user) {
  return {
    user,
    type: SIGN_UP_REQUEST,
  };
}

export function signUpResponse(response) {
  return {
    type: SIGN_UP_SUCCESS,
    response
  }
}

export function signUpError(error) {
  return {
    type: SIGN_UP_ERROR,
    error
  }
}
