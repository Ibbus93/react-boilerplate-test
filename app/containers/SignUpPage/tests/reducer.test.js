/**
 * How to test Reducers
 *
 * Reducers are pure functions, hence the expected result is only the
 * immer produce with the result we are looking for
 *
 * The alternative is to use snapshots, how it could be seen in the 2nd and 3rd test
 */

import produce from 'immer';

import signUpReducer from '../reducer';
import { signUpRequest, signUpResponse, signUpError } from '../actions';
import { defaultState, response, error } from './mocks';

/* eslint-disable default-case, no-param-reassign */
describe('signUpReducer', () => {
  let state;

  beforeEach(() => {
    state = defaultState;
  });

  it('returns the initial state', () => {
    const expectedResult = state;
    expect(signUpReducer(undefined, {})).toEqual(expectedResult);
  });

  it('should handle the signUpRequest action correctly', () => {
    // const expectedResult = produce(state, draft => {
    //   draft.loading = true;
    //   draft.isError = false;
    // });

    // expect(signUpReducer(state, signUpRequest())).toEqual(expectedResult);

    expect(signUpReducer(state, signUpRequest())).toMatchSnapshot();
  });

  it('should handle the signUpResponse action correctly', () => {
    // const expectedResult = produce(state, draft => {
    //   draft.loading = false;
    //   draft.response = response;
    // });

    // expect(signUpReducer(state, signUpResponse(response))).toEqual(expectedResult);

    expect(signUpReducer(state, signUpResponse(response))).toMatchSnapshot();
  });

  it('should handle the signUpError action correctly', () => {
    const expectedResult = produce(state, draft => {
      draft.loading = false;
      draft.isError = true;
      draft.error = error;
    });

    expect(signUpReducer(state, signUpError(error))).toEqual(expectedResult);
  });
});
