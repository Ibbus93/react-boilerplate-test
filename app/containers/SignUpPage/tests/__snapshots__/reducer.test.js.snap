// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUpReducer should handle the signUpRequest action correctly 1`] = `
Object {
  "error": null,
  "isError": false,
  "loading": true,
  "response": null,
}
`;

exports[`signUpReducer should handle the signUpResponse action correctly 1`] = `
Object {
  "error": null,
  "isError": false,
  "loading": false,
  "response": Object {
    "data": Object {
      "email": "gino@bartali.it",
      "name": "Gino",
      "surname": "Bartali",
      "username": "bartali",
    },
  },
}
`;
